vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_info.cc

```diff
@@ -0,0 +1,42 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+#include <vector>
+
+#include "bat/ads/internal/ml/ml_util.h"
+#include "bat/ads/internal/ml/pipeline/pipeline_info.h"
+
+namespace ads {
+namespace ml {
+namespace pipeline {
+
+PipelineInfo::PipelineInfo() = default;
+
+PipelineInfo::PipelineInfo(const PipelineInfo& pinfo) {
+  version = pinfo.version;
+  timestamp = pinfo.timestamp;
+  locale = pinfo.locale;
+  linear_model = pinfo.linear_model;
+  transformations = GetTransformationVectorCopy(pinfo.transformations);
+}
+
+PipelineInfo::~PipelineInfo() = default;
+
+PipelineInfo::PipelineInfo(const uint16_t& version,
+                           const std::string& timestamp,
+                           const std::string& locale,
+                           const TransformationVector& new_transformations,
+                           const model::Linear& linear_model)
+    : version(version),
+      timestamp(timestamp),
+      locale(locale),
+      linear_model(linear_model) {
+  transformations = GetTransformationVectorCopy(new_transformations);
+}
+
```

Comment:
@tmancey for sake of consistency, do we want to overload `=` and `!=` here?

Reply:
yes please as good for tests

Reply:
As we discussed, it's impossible to overload any method or operator for `std::vector` of `std::unique_ptr` objects.